tests/ip/test_ip_packet.py

```diff
@@ -668,3 +676,59 @@ def test_drop_ip_packet_with_wrong_0xffff_chksum(self, duthosts, enum_rand_one_p
         pytest_assert(max(tx_drp, tx_err) <= self.PKT_NUM_ZERO,
                       "Dropped {} packets in tx, not in expected range".format(tx_err))
+
+    def test_drop_l3_ip_packet_non_dut_mac(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname,
+                                           ptfadapter, common_param, skip_traffic_test):  # noqa F811
+        # GIVEN a random normal ip packet, and random dest mac address
+        # WHEN send the packet to DUT with dst_mac != ingress_router_mac to a layer 3 interface
+        # THEN DUT should drop it and add drop count
+        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+        (peer_ip_ifaces_pair, rif_rx_ifaces, rif_support, ptf_port_idx,
+         pc_ports_map, ptf_indices, ingress_router_mac) = common_param
+
+        dst_mac = TestIPPacket.random_mac()
+        while dst_mac == ingress_router_mac:
+            dst_mac = TestIPPacket.random_mac()
+
+        pkt = testutils.simple_ip_packet(
+            eth_dst=dst_mac,
+            eth_src=ptfadapter.dataplane.get_mac(0, ptf_port_idx),
+            ip_src=peer_ip_ifaces_pair[0][0],
+            ip_dst=peer_ip_ifaces_pair[1][0])
+
+        out_rif_ifaces, out_ifaces = TestIPPacket.parse_interfaces(
+            duthost.command("show ip route %s" % peer_ip_ifaces_pair[1][0])["stdout_lines"], pc_ports_map)
+
+        duthost.command("portstat -c")
+        if rif_support:
+            duthost.command("sonic-clear rifcounters")
+        ptfadapter.dataplane.flush()
+
+        testutils.send(ptfadapter, ptf_port_idx, pkt, self.PKT_NUM)
+        time.sleep(5)
+
+        portstat_out = parse_portstat(duthost.command("portstat")["stdout_lines"])
+        if rif_support:
+            rif_counter_out = TestIPPacket.parse_rif_counters(
+                duthost.command("show interfaces counters rif")["stdout_lines"])
+
+        # rx_ok counter to increase to show packets are being received correctly at layer 2
+        # rx_drp or rx_err counter to increase to show packets are being dropped
+        # tx_ok, tx_drop, tx_err counter to zero to show no packets are being forwarded
+        rx_ok = int(portstat_out[peer_ip_ifaces_pair[0][1][0]]["rx_ok"].replace(",", ""))
+        rx_drp = int(portstat_out[peer_ip_ifaces_pair[0][1][0]]["rx_drp"].replace(",", ""))
+        rx_err = int(rif_counter_out[rif_rx_ifaces]["rx_err"].replace(",", "")) if rif_support else 0
+        tx_ok = TestIPPacket.sum_ifaces_counts(portstat_out, out_ifaces, "tx_ok")
+        tx_drp = TestIPPacket.sum_ifaces_counts(portstat_out, out_ifaces, "tx_drp")
+        tx_err = TestIPPacket.sum_ifaces_counts(rif_counter_out, out_rif_ifaces, "tx_err") if rif_support else 0
+
+        if skip_traffic_test is True:
+            return
+        pytest_assert(rx_ok >= self.PKT_NUM_MIN,
+                      "Received {} packets in rx, not in expected range".format(rx_ok))
+        pytest_assert(max(rx_drp, rx_err) >= self.PKT_NUM_MIN,
```

Comment:
why we need to compare rif error and intf drop?

Reply:
Good question. Was checking rx_err to be consistent with other test cases in the same module.
However,  the packet itself shouldn't enter into Layer 3 forwarding in this case. Both cisco and 7260(broadcom) just increased rx_drp counter, not rx_rif_ counter. Will remove the rx_err check for now, if other platform has different behavior, we can add platform specific check when it happens.